static/app/views/releases/thresholdsList/index.tsx

```diff
@@ -168,32 +174,29 @@ function ReleaseThresholdList({}: Props) {
 
   const thresholdsByProject: {[key: string]: Threshold[]} = useMemo(() => {
     const byProj = {};
+
     filteredThresholds.forEach(threshold => {
-      const projId = threshold.project.id;
-      if (!byProj[projId]) {
-        byProj[projId] = [];
-      }
-      byProj[projId].push(threshold);
+      const selectedProject = selection.projects[0] !== -1 ? selection.projects[0] : null;
+      const projId = threshold.project.id ?? selectedProject;
+      (byProj[projId] ??= []).push(threshold);
```

Comment:
I'm a bit confused what this part does. It seems to be returning a mapping of project id to threshold. When is it the case that a threshold does not have a project id attached to it, and we need to fallback to the selection?

Reply:
heh - this is just newer javascript syntax.

`(byProj[projId] ??= []).push(threshold);`
is the same as:
```
if (!byProj[projId]) {
  byProj[projId] = []
}
byProj[projId].push(threshold)
```

Reply:
Ohhh my bad - i'm so used to getting questions about that syntax i always have it ready to go in a paste buffer in my brain.

The reason for the `projId` with the selected projects stuff is because the result we get from a MetricRule only returns the project slug. Rather than making `n` requests to fetch the project information, i updated the hash to use the id or hash (prefer id) to lookup.

I more or less am tricking this layer here: https://github.com/getsentry/sentry/pull/67637/files#diff-c42088895c5006654b18b6542c0f26c8214ec4f7ddaf80347903951c7add1ddaR64 -- allowing me to avoid the `n` extra requests.

Reply:
ahhh that makes sense! thanks for the explanation!